src/token/erc20/extensions/erc20votes.cairo

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.11.0 (token/erc20/extensions/erc20votes.cairo)
+
+use core::hash::HashStateExTrait;
+use hash::{HashStateTrait, Hash};
+use openzeppelin::utils::cryptography::snip12::{OffchainMessageHash, StructHash, SNIP12Metadata};
+use poseidon::PoseidonTrait;
+use starknet::ContractAddress;
+
+/// # ERC20Votes Component
+///
+/// The ERC20Votes component tracks voting units from ERC20 balances, which are a measure of voting power that can be
+/// transferred, and provides a system of vote delegation, where an account can delegate its voting units to a sort of
+/// "representative" that will pool delegated voting units from different accounts and can then use it to vote in
+/// decisions. In fact, voting units MUST be delegated in order to count as actual votes, and an account has to
+/// delegate those votes to itself if it wishes to participate in decisions and does not have a trusted representative.
```

Comment:
Very nice description! Once the docs are merged, I think it'd be especially helpful to link them here